documentation/specs/Notification.md

```diff
@@ -0,0 +1,567 @@
+
+### Toast Use Cases
+
+- Inform user of a completed action.
+- Inform user that an asynchronous process has been triggered based on their action.
+
+### Alert Use Cases
+
+- Inform user of system level errors such as a failed network request.
+- Inform user of any critical information that requires their attention.
+
+### Features
+
+- Notifications are interacted with through the use of a `useNotification` hook.
+- Toasts have ARIA `role="status"` and alerts have ARIA `role="alert"`.
+- The `message` prop is of type `ReactNode`.
+
+### Risks and Challenges
+
+- Unlike other components, `Notification` components are unique in that they may be quite opinionated with regards to how they display and the kind of control they expose to consumers.
+- Building for the shipper's experience at EasyPost according to internal specs while maintaining a pleasant DX for outside consumers.
+- Allowing consumers to customize placement.
+
+### Prior Art
+
+- [Paste `<Alert />`](https://paste.twilio.design/components/alert)
+- [Paste `<Toast />`](https://paste.twilio.design/components/toast)
+- [Spectrum `<Toast />`](https://react-spectrum.adobe.com/react-spectrum/Toast.html)
+
+---
+
+## Design
+
+### API
+
+```typescript
+import type { AriaToastProps } from "@react-aria/toast";
+import type { QueuedToast } from "@react-stately/toast";
+
+export type NotificationType = "alert" | "toast";
+
+export type NotificationStatus =
+  | "promotional"
+  | "success"
+  | "neutral"
+  | "error"
+  | "warning";
+
+export type NotificationProps = {
+  /** Notification type */
+  type: NotificationStatus;
+  /** Notification message */
+  message: ReactNode;
+  /** Callback function when dismissing alerts */
+  onDismiss?: () => void;
+  /**
+   * Notification status
+   * @default "success"
+   */
+  status?: NotificationStatus;
+  /**
+   * Notification will render with left aligned status icon
+   * @default false
+   */
+  noIcon?: boolean;
+};
+
+export type NotificationStateProps = AriaToastProps<NotificationProps> & {
+  /**
+   * The `toast` name here is an artifact of react-stately, not an Easy UI `toast`
+   * it is more appropriate to think of this as a `notification`.
+   */
+  toast: QueuedToast<NotificationProps>;
+  /**
+   * Holds the state for notifications and the functions that directly
+   * interact with the queue object.
+   */
+  state: NotificationState<NotificationProps>;
+};
+
+/** The methods below are accessible on the object returned by the `useNotification` hook */
+export type NotificationState<NotificationProps> =
+  ToastState<NotificationProps> & {
+    /** Shows promotional color status toast */
+    showPromotionalToast(content: NotificationProps): void;
```

Comment:
**thought:** since each method defines both the `type` and the `status`, i guess we would want a type that doesn't include those properties to be used on `content` instead. something like the below.

```
type NotificationContent = Omit<NotificationProps, "type" | "status">;

showPromotionalToast(content: NotificationContent)
```